core/migration_external_user_id.go

```diff
@@ -0,0 +1,141 @@
+func (app *Application) MigrateScoreExternalUserIDs(orgID string, appID string, batchSize int) error {
+	log.Println("Starting Score External User ID migration...")
+	log.Printf("Parameters: orgID=%s, appID=%s, batchSize=%d", orgID, appID, batchSize)
+
+	// Validate batch size
+	if batchSize <= 0 {
+		batchSize = DefaultExternalUserIDMigrationBatchSize
+		log.Printf("Invalid batch size, using default: %d", batchSize)
+	}
+
+	successCount := 0
+	errorCount := 0
+	skippedCount := 0
+	total := 0
+	offset := 0
+
+	// Process in batches to avoid overwhelming the Core BB API and memory
+	for {
+		// Fetch a batch of scores using limit and offset
+		batch, err := app.storage.FindScoresWithoutExternalUserID(orgID, appID, &batchSize, &offset)
+		if err != nil {
+			return fmt.Errorf("error fetching scores: %v", err)
+		}
+
+		// If no more scores, we're done
+		if len(batch) == 0 {
+			break
+		}
+
+		total += len(batch)
+		log.Printf("Processing batch %d-%d (fetched %d scores)", offset+1, offset+len(batch), len(batch))
+
+		// Collect unique account IDs from scores in this batch
+		accountIDs := []string{}
+		for _, score := range batch {
+			if score.UserID == "" {
+				log.Printf("Score %s has no UserID (Core BB account ID), skipping", score.ID)
+				skippedCount++
+				continue
+			}
+			accountIDs = append(accountIDs, score.UserID)
+		}
+
+		if len(accountIDs) == 0 {
+			log.Printf("No valid account IDs in this batch, skipping")
+		} else {
+			log.Printf("Fetching Core BB accounts for %d account IDs", len(accountIDs))
+
+			// Look up all AmgUUIDs from Core BB in one call
+			coreAccounts, err := app.corebb.RetrieveCoreUserAccountByCriteria(accountIDs, &appID, &orgID)
+			if err != nil {
+				log.Printf("Error retrieving Core accounts for batch: %v", err)
+				errorCount += len(accountIDs)
+			} else {
+				log.Printf("Retrieved %d Core BB accounts", len(coreAccounts))
+
+				// Build a map of account ID -> amg_uuid for quick lookup
+				accountIDToAmgUUID := make(map[string]string)
+				for _, account := range coreAccounts {
+					amgUUID := account.GetAmgUUID()
+					if account.ID != "" && amgUUID != "" {
+						accountIDToAmgUUID[account.ID] = amgUUID
+					}
+				}
+
+				// Update scores with their corresponding AmgUUIDs
+				for _, score := range batch {
+					if score.UserID == "" {
+						continue
+					}
+
+					amgUUID, found := accountIDToAmgUUID[score.UserID]
+					if !found {
+						log.Printf("No Core account found for account ID: %s", score.UserID)
+						errorCount++
+						continue
+					}
+
+					err = app.storage.UpdateScoreExternalUserID(score.ID, amgUUID)
+					if err != nil {
+						log.Printf("Error updating score %s: %v", score.ID, err)
+						errorCount++
+						continue
+					}
+
+					log.Printf("✓ Score %s: UserID=%s → external_user_id=%s",
+						score.ID, score.UserID, amgUUID)
+					successCount++
+				}
+			}
+		}
+
+		// Move to next batch
+		offset += batchSize
```

Comment:
There is an issue with the logic here for the batching offset. Note that we are updating the score documents as we iterate through the batches, meaning that these documents will be removed from the results the next time we load the next batch. 

Effectively, I think this means that we should not be changing the offset at all since we should be removing the previous batch each time before we move on to the next one.